framework/helpers/docker.py

```diff
@@ -0,0 +1,295 @@
+import datetime
+import math
+import pathlib
+import queue
+import threading
+
+import absl
+import grpc
+import mako.template
+
+import docker
+from protos.grpc.testing import messages_pb2
+from protos.grpc.testing import test_pb2_grpc
+from protos.grpc.testing.xdsconfig.control_pb2 import StopOnRequestRequest
+from protos.grpc.testing.xdsconfig.control_pb2 import UpsertResourcesRequest
+from protos.grpc.testing.xdsconfig.service_pb2_grpc import (
+    XdsConfigControlServiceStub,
+)
+
+
+class Bootstrap:
+    def __init__(self, base: str, ports: list[int], host_name: str):
+        self.base = pathlib.PosixPath(base)
+        self.host_name = host_name
+        self.ports = ports
+        self.mount_dir: pathlib.Path = None
+        self.MakeWorkingDir(self.base)
+        # Use Mako
+        template = mako.template.Template(filename="templates/bootstrap.json")
+        file = template.render(
+            servers=[f"{self.host_name}:{port}" for port in self.ports]
+        )
+        destination = self.get_mount_dir() / "bootstrap.json"
+        with open(destination, "w", encoding="utf-8") as f:
+            f.write(file)
+            absl.logging.debug("Generated bootstrap file at %s", destination)
+
+    def MakeWorkingDir(self, base: str):
```

Comment:
Here and elsewhere: use `snake_case`. The only exception from it is classes derived from unittest.Testcase that require to override methods like `setUpClass`, etc.